daemon/algod/api/algod.oas2.json

```diff
@@ -4441,6 +4441,33 @@
           }
         }
       }
+    },
+    "BlockLog": {
+      "description": "An array of logs for a given application ID and outer transaction ID",
```

Comment:
I think you want a slightly different schema for your response.  I think you have:
```
[ { "txid": BC12433, "application-index": 345, "items": [ a bunch of strings] },
  { "txid": FA62233, "application-index": 27364, "items": [ a bunch of strings] },
  .
  .
]
```
And I'm not sure if you want one of these objects for _every_ top-level transaction, or only if the top-level txn is an app call.

Regardless, I think you need hierarchy, in order to convey that some of those logs were written by inner apps with different app-ids.  This might work.
```
[ {txid: "FA56E" logs: [{app-id: 123, str1}, {app-id:233: str2}...]},
   {txid: "A6B5E" logs: [{app-id: 311, str1}, {app-id:111: str2}...]},
]
```
Here a top-level app called an inner app, and each logged one thing.

Reply:
I think the most common use case of this endpoint will be for parsing events. For events, you typically don't care about how it was called. Any necessary information will be encoded in the event itself. Having a nested structure will make it a bit harder to parse all the logs and see if an event you care about happened.

If we want to include "location" in the response then I would propose we add a `LoggedAt` field that resembles `FailedAt` from the simulate response:

> FailedAt If present, indicates which transaction in this group caused the failure. This array represents the path to the failing transaction. Indexes are zero based, the first element indicates the top-level transaction, and successive elements indicate deeper inner transactions.

Reply:
I like the current schema. It matches the "proposed solution" from #5807 which I helped define and was happy with.

My opinion on edge cases are:
* As @jannotti pointed out, txns with no logs should be filtered out (right now I think they're still present with an empty logs array)
* If a top-level app invokes the same app multiple times as an inner, I think the inners ought to be separate objects. E.g.
```json
[
  { "txid": "TOPLVL", "application-index": 123, "items": ["..."] },
  { "txid": "TOPLVL", "application-index": 123, "items": ["..."] }
]
```

Reasoning for the second bullet is that this is essentially the same as two top-level app calls to the same app in the same group, and in that case they would be separate objects. Plus it seems "nice" that each object represents a single transaction.

Reply:
> I think that you'll naturally get your second bullet point because each inner is processed independently right? There's no collection by id going on.

With the current code yes that naturally falls out. For a moment I was questioning what makes "the most sense" though, so I was considering alternatives.

Reply:
And to @joe-p's point, it would be easy to add a `FailedAt`-like field to point to exactly which inner was responsible for those logs. But in my opinion is that's not necessary based on our current expectations of how events will be processed. Can always revisit if there's a need for it though.